packages/flutter_test/lib/src/instrumentation.dart

```diff
@@ -0,0 +1,173 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' as ui;
+
+import 'package:flutter/gestures.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+import 'test_pointer.dart';
+
+typedef Point SizeToPointFunction(Size size);
+
+/// Helper class for flutter tests providing event dispatch.
+///
+/// This class provides hooks for accessing the rendering tree and dispatching
+/// fake tap/drag/etc. events.
+class Instrumentation {
+    Instrumentation()
+      : binding = WidgetFlutterBinding.ensureInitialized()
+    {}
+
+    final WidgetFlutterBinding binding;
+
+    List<Layer> _layers(Layer layer) {
+      List<Layer> result = <Layer>[layer];
+      if (layer is ContainerLayer) {
+        ContainerLayer root = layer;
+        Layer child = root.firstChild;
+        while (child != null) {
+          result.addAll(_layers(child));
+          child = child.nextSibling;
+        }
+      }
+      return result;
+    }
+    List<Layer> get layers => _layers(binding.renderView.layer);
```

Comment:
getters should not have hidden O(N) algorithms. :-)
Should probably be a method.